### DISCUSSION TITLE
Performance Stats for Executing Rules Engine (3.5.0)

### DISCUSSION THREAD

**Discussion by AmithDotKota** [microsoft/RulesEngine#459] [Q&A]
Do we have any stats available for executing ExecuteAllRulesAsync ? We are using like 21 workflows. Each workflow has like 70 global parameters and there are about 10 rules.

Most of the calls to ExecuteAllRulesAsync have no latency, but under production load, we are seeing a latency of more than 10 seconds. Has anyone noticed this ? Please advise if we have to do anything different for this.

**Comment by abbasc52** [comment]
@AmithDotKota Is it possible for you to analyze the CPU and Memory when this happens? (It can be a memory leak issue).
Also if you can provide details on how you are using it, we can guide you better.
There is also benchmarking project in the repo, you can edit it to test your scenario and evaluate the performance.
